docs/framework/react/guide/custom-link.md

```diff
@@ -2,91 +2,160 @@
 title: Custom Link
 ---
 
-While repeating yourself can be fine in many cases, you may find yourself repeating yourself too much. Sometimes you may want to make cross cutting components with extra behavior or styles. Or sometimes you may also use third party libraries and combine it with TanStack Router's type safety
+While repeating yourself can be acceptable in many situations, you might find that you do it too often. At times, you may want to create cross-cutting components with additional behavior or styles. You might also consider using third-party libraries in combination with TanStack Router’s type safety.
 
-## `createLink` for cross cutting concerns
+## `createLink` for cross-cutting concerns
 
 `createLink` creates a custom `Link` component with the same type parameters as `Link`. This means you can create your own component which provides the same type safety and typescript performance as `Link`.
 
-If you want to create a component which wraps `Link` with some additional styles or behavior then you can do so with the following
+### Basic example
+
+If you want to create a basic custom link component, you can do so with the following:
 
 ```tsx
 import * as React from 'react'
-import { Link, createLink } from '@tanstack/react-router'
+import { createLink, LinkComponent } from '@tanstack/react-router'
+
+interface BasicLinkProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
```

Comment:
If we allow all the `<a>` props via `React.AnchorHTMLAttributes<HTMLAnchorElement>`, then the `href` prop would be allowed:
```tsx
<CreatedLinkComponent href="/path" to="/path" />
```
This would be allowed, which is probably not what we want, no?

Tangentially, I've also noticed that the `disabled` prop is passed, which is not a valid `<a>` prop.

Reply:
You're right, but I'm unsure how deeply it makes sense to example such things. The user can easily omit the `href` from the props in this case but it always depends on the component. 

I'm uncertain how to handle the `disabled` prop with a custom component.